### ISSUE TITLE
readme.txt on self hosted plugin

### ISSUE THREAD

**Issue by jwkellyiii** [YahnisElsts/plugin-update-checker#301]
Your instructions do not mention the use of a `readme.txt` file on the Self Hosted Plugin or Theme instructions. Does that mean if we are going to have a self hosted plugin or theme (not in GitHub or BitBucket) that this code will not read a `readme.txt` file? Does the `readme.txt` file only get read if we have GitHub or BitBucket integration?

**Comment by YahnisElsts** [comment]
That's correct, PUC will not automatically read the `readme.txt` file of self-hosted plugins and themes. If you want it to use information from `readme.txt` - such as the changelog, description, and so on - you need to include it in the JSON metadata.

You can automate that process by using [`wp-update-server`](https://github.com/YahnisElsts/wp-update-server) to provide update information to PUC. Among other things, `wp-update-server` will parse the `readme.txt` file and include the relevant sections in its output.
